pandas/core/computation/common.py

```diff
@@ -26,3 +26,20 @@ def result_type_many(*arrays_and_dtypes):
     except ValueError:
         # we have > NPY_MAXARGS terms in our expression
         return reduce(np.result_type, arrays_and_dtypes)
+    except TypeError:
+        from pandas.core.dtypes.cast import find_common_type
+        from pandas.core.dtypes.common import is_extension_array_dtype
+
+        arr_and_dtypes = list(arrays_and_dtypes)
+        ea_dtypes, non_ea_dtypes = [], []
+        for arr_or_dtype in arr_and_dtypes:
+            if is_extension_array_dtype(arr_or_dtype):
+                ea_dtypes.append(arr_or_dtype)
+            else:
+                non_ea_dtypes.append(arr_or_dtype)
+
+        if non_ea_dtypes:
+            np_dtype = np.result_type(*non_ea_dtypes)
```

Comment:
Could this hit the `except ValueError:` as above?

Reply:
Not totally sure, we don't have a test hitting the ValueError above, but added it down here as well